java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink.java

```diff
@@ -68,12 +74,17 @@ void addOkResult(int numberOfActions) {
             blockingQueue.add(new EsWriteResultResp(numberOfActions));
         }
 
-        void addWriteTask() {
+        void addWriteTask() throws InterruptedException {
             taskCount++;
             EsWriteResultResp esWriteResultResp;
             while (true) {
-                if ((esWriteResultResp = this.blockingQueue.poll()) != null) {
+                if ((esWriteResultResp = this.blockingQueue.poll(10, TimeUnit.MILLISECONDS))
+                        != null) {
                     checkEsWriteResultResp(esWriteResultResp);
+                }
+
+                if (taskCount >= maxTaskNum) {
```

Comment:
ES bulkprocessoer seems to be built-in support for batching and concurrency limit. Should we use `setBulkActions(maxTaskNum)` instead?

Reply:
It looks like setBulkActions is used to set the batch size, and this is to set the parallelism of writing to es. They seem to control two things

Reply:
IIUC, `setConcurrentRequests` is used to control the concurrency and `setBulkActions` is used to control the batch size. We use `setConcurrentRequests(1)` in es sink so I guess the concurrency is always 1. That is why I thought the `maxTaskNum` is used to control batch size.

Reply:
fix